collection_requirements.rst

```diff
@@ -262,11 +262,16 @@ ways:
                <https://opensource.org/licenses/BSD-2-Clause>`_ license to make it possible for
                third-party modules which are licensed incompatibly with the GPLv3 to use them.
                Please consider this use case when licensing your own ``module_utils``.
-:All other code: All other code must be under the `GPL-3.0-or-later
-                 <https://www.gnu.org/licenses/gpl-3.0-standalone.html>`_.  These plugins are run
-                 inside of the Ansible controller process which is licensed under the GPLv3+ and
-                 often must import code from the controller.  For these reasons, the GPLv3+ must be
-                 used.
+:All other code in ``plugins/``: All other code in ``plugins/`` must be under the `GPL-3.0-or-later
+                                 <https://www.gnu.org/licenses/gpl-3.0-standalone.html>`_.  These plugins
+                                 are run inside of the Ansible controller process which is licensed under
+                                 the GPLv3+ and often must import code from the controller.  For these
+                                 reasons, the GPLv3+ must be used.
+:All other code: Code outside ``plugins/``, in particular unit tests, must be licensed with
```

Comment:
Sorry for not batching up my feedback so you could apply it all at once.

```suggestion
:All other code: Code outside ``plugins/``, in particular unit tests for ``modules`` and ``module_utils``, must be licensed with
```

Reply:
No problem about batching :)

I'm not sure why unit tests for modules and module_utils should be singled out here. Other unit tests also have to be licensed with a free software license (though very likely with GPLv3+, at least the way I interpret it, since IANAL).

Reply:
Well, wouldn't the tests for the other plugins import the code from the other plugins that have to be GPL-licensed?

Reply:
Yes, but (there's always a but ;) ) you can also have Python files there that do not import non-module/module_utils plugins. For example you could have a Apache 2.0 licensed file that provides some utility that is only used by a test for an inventory plugin, and thus located right next to it. Or you can have generic unit test helpers in `tests/unit/` outside of `tests/unit/plugins/modules/` and `tests/unit/plugins/module_utils/` (like the mock import helper) that are licensed under another license than GPLv3+.

Reply:
Maybe it makes sense to phrase it like so?

```
:All other code: Code outside ``plugins/``, in particular unit tests [for ``modules`` and ``module_utils``], may be licensed under
                 another free software license that is compatible with the
                 `GPL-3.0-or-later <https://www.gnu.org/licenses/gpl-3.0-standalone.html>`_,
                 provided that such code that does not import any code that is licensed under
                 the ``GPL-3.0-or-later``.
```

(The bracketed part would either be removed or included without the brackets)

I don't feel strongly either way, but I think my suggestion might be clearer.

Reply:
Hmm, I think this is only easier to understand for folks who know that code importing GPLv3+ licensed code needs to be GPLv3+ licensed as well. How about adding something like "Code satisfying the latter must be licensed under the `GPL-3.0-or-later`."?

Reply:
I've edited the suggestion.

Reply:
Applied as 0a1b0f352485e1ddbbac944130499b2b2a205440 (without the square brackets).